frontend/src/store/modules/user.ts

```diff
@@ -62,21 +65,39 @@ export default {
       user: firebase.User | null
     ) {
       commit("setCurrentUser", user);
-      if (user) {
-        await dispatch(
-          "updateUserRoles",
-          (await user.getIdTokenResult()).claims.roles
-        );
-      } else {
-        await dispatch("updateUserRoles", null);
-      }
+      dispatch("updateUserRoles", await dispatch("getUserRoles"));
     },
     updateUserRoles(
       { commit, getters }: ActionContext<any, any>,
       roles: { [key: string]: any } | null
     ) {
       commit("setUserRoles", roles);
       getters.ability.update(defineAbilities());
+    },
+    async getUserRoles(
+      { state }: ActionContext<any, any>,
+      refreshToken = false
+    ) {
+      if (!state.currentUser) return null;
+      return (await state.currentUser.getIdTokenResult(refreshToken)).claims
+        .roles;
+    },
+    handleMetaData({ state, dispatch }: ActionContext<any, any>) {
```

Comment:
The name is confusing. At first I thought that it actually handles the metadata change, similar to `handleUser`.

Can't we move all this code into the `setCurrentUser` mutation? Because we need to re-hook only when the user changes. I see no async code here, so why not to move?